### PR TITLE
[smol] Hook up mailing list response submission

### PR DESCRIPTION
**Pull request by jtaaa**
[🔌](https://docs.google.com/spreadsheets/d/16PEbh4LFsb1klDX-W6kp47_p9acEKVIWNBz-TTDXcDc/edit#gid=0)

### CODE REVIEW

File: client/src/utils/network/errorMessages.ts [hunk 1]
```diff
@@ -4,3 +4,5 @@ export const THIRD_PARTY_AUTH = 'Could not authenticate with third party';
 export const FETCH_USER = 'Could not fetch user';
 export const FETCH_USERS = 'Could not fetch all users';
 export const LOGOUT = 'Could not log user out';
+
+export const MAILING_LIST = 'Could not submit mailing list response';
```
Comment (chriskrogh) [inline client/src/utils/network/errorMessages.ts:8]:
```suggestion
export const MAILING_LIST = 'Could not submit mailing list request';
```

File: client/src/utils/network/endpoints.ts [hunk 2]
```diff
@@ -1,3 +1,4 @@
 export enum APIRoutes {
   USER = '/api/user',
+  MAILING_LIST = '/api/mailingList',
```
Comment (chriskrogh) [inline client/src/utils/network/endpoints.ts:3]: Yeaaaaaaaa
We need something like this for the backend bro

File: client/src/types/mailingList.ts [hunk 3]
```diff
@@ -0,0 +1,5 @@
+export type MailingListResponse = {
```
Comment (chriskrogh) [inline client/src/types/mailingList.ts:1]:
```suggestion
export type MailingListRequest = {
```

File: client/src/components/forms/MailingList/index.tsx [hunk 4]
```diff
@@ -85,6 +86,19 @@ function MailingListForm({ classes }: WithStyles<typeof styles>) {
...
+        email,
+        interestedEvents: events.map((event) => event.title),
+        otherFeedback: feedback,
+      });
+      goRight();
+    } catch (err) {
+      console.error(err);
+    }
+  };
+
```
Comment (chriskrogh) [inline client/src/components/forms/MailingList/index.tsx:101]: What a beauty😍

**Comment by jtaaa** [comment]
Darn yeah I actually meant response as in the user's response to the mailing list form but now I see how that is v confusing

File: client/src/components/forms/MailingList/index.tsx (hunk 4 above)
Comment (jtaaa) [reply client/src/components/forms/MailingList/index.tsx:101]: If for any reason it fails the user can open up their console to see why 😏 
#UX

**Pull request merged**